projects/cm_mcu/CommandLineTask.c

```diff
@@ -161,16 +161,99 @@ static BaseType_t snapshot(int argc, char **argv, char *m)
   _Static_assert(sizeof(snapshot_t) == 32, "sizeof snapshot_t");
   int copied = 0;
   int page = strtol(argv[1], NULL, 10); // which LGA08D
+  if (page == 0) {
```

Comment:
are you using the fact that if the `strtol` fails here the returned value is 0? This is correct but rather confusing from a program flow point of view. The confusion comes from the fact that `strtol` is really hard to use correctly; see e.g. [this discussio](https://stackoverflow.com/questions/14176123/correct-usage-of-strtol)

A more clean/less clever use might be the following:
1. first check if argv[1] is 'all' via `strncmp`
2. if not do the `strtol` to figure out what page
Also I think we should not print out the snapshot register when we do the clearing; instead, we should just have a clear command that says 'resetall' and then reset all the snapshot registers and not print the output. In fact it would be hard to look at all these outputs if they get dumped all at once. (It also avoids the large amount of code duplication introduced right now with this PR)